File: python/Code/Submission.py
from Student import Student
from Assignment import Assignment

# Kelas untuk merepresentasikan data sebuah submission (pengumpulan tugas)
class Submission:
    # Konstruktor untuk membuat objek Submission baru
    def __init__(self, student, date, assignment, grade):
        self.__student = student
        self.__date = date
        self.__assignment = assignment
        self.__grade = grade

    # --- Getters ---
    # Mengambil objek mahasiswa
    def getStudent(self):
        return self.__student
        
    # Mengambil objek tugas
    def getAssignment(self):
        return self.__assignment

    # Mengambil tanggal submission
    def getSubmissionDate(self):
        return self.__date

    # Mengambil nilai
    def getGrade(self):
        return self.__grade

    # --- Setters ---
    # Mengubah mahasiswa yang bersangkutan
    def setStudent(self, student):
        self.__student = student
        
    # Mengubah tugas yang bersangkutan
    def setAssignment(self, assignment):
        self.__assignment = assignment

    # Mengubah tanggal submission
    def setSubmissionDate(self, date):
        self.__date = date

    # Mengubah atau memberikan nilai
    def setGrade(self, grade):
        self.__grade = grade

File: java/Code/Submission.java
// Kelas untuk merepresentasikan data sebuah submission (pengumpulan tugas)
public class Submission {
    // Atribut
    private Student student;
    private String submissionDate;
    private Assignment assignment;
    private int grade;

    // Konstruktor untuk membuat objek Submission baru
    public Submission(Student student, String submissionDate, Assignment assignment, int grade) {
        this.student = student;
        this.submissionDate = submissionDate;
        this.assignment = assignment;
        this.grade = grade;
    }

    // Mengambil objek mahasiswa
    public Student getStudent() {
        return student;
    }

    // Mengambil tanggal submission
    public String getSubmissionDate() {
        return submissionDate;
    }

    // Mengambil objek tugas
    public Assignment getAssignment() {
        return assignment;
    }

    // Mengambil nilai
    public int getGrade() {
        return grade;
    }
    
    // Mengubah mahasiswa yang bersangkutan
    public void setStudent(Student student) {
        this.student = student;
    }

    // Mengubah tanggal submission
    public void setSubmissionDate(String submissionDate) {
        this.submissionDate = submissionDate;
    }

    // Mengubah tugas yang bersangkutan
    public void setAssignment(Assignment assignment) {
        this.assignment = assignment;
    }

    // Mengubah atau memberikan nilai
    public void setGrade(int grade) {
        this.grade = grade;
    }
}
